documenters_aggregator/spiders/pbcc.py

```diff
@@ -0,0 +1,198 @@
+        """
+        `parse` should always `yield` a dict that follows the `Open Civic Data
+        event standard <http://docs.opencivicdata.org/en/latest/data/event.html>`_.
+
+        Change the `_parse_id`, `_parse_name`, etc methods to fit your scraping
+        needs.
+        """
+        for cal_day in response.css('td.calday2'):
+            cal_number = cal_day.xpath("./div[contains(@class, 'calnumber')]/text()").extract_first()
+            if not cal_number:
+                continue
+            cal_date = date(self.calendar_date.year, self.calendar_date.month, int(cal_number))
+            for item in cal_day.xpath("./div[not(contains(@class, 'calnumber'))]"):
+                # Ignore holidays added to calendar
+                if item.css('a b::text').extract_first().strip() != 'Holiday':
+                    yield {
+                        '_type': 'event',
+                        'id': self._parse_id(item),
+                        'name': self._parse_name(item, cal_date),
+                        'description': self._parse_description(item),
+                        'classification': self._parse_classification(item),
+                        'start_time': self._parse_start(item, cal_date),
+                        'end_time': None,
+                        'all_day': self._parse_all_day(item),
+                        'status': self._parse_status(item),
+                        'location': self._parse_location(item),
+                        'sources': self._parse_sources(item)
+                    }
+
+        # Add 30 days to the current date, stop if more than 180 days (~6 months) ahead
+        self.calendar_date += timedelta(days=30)
+        if (self.calendar_date - datetime.now()).days <= 180:
+            yield self._parse_next(response)
+
+    def _parse_next(self, response):
+        """
+        Get next page. You must add logic to `next_url` and
+        return a scrapy request.
+
+        The parse method updates the calendar_date property,
+        so create a query param from that.
+        """
+        next_url = '{}?myDate={}'.format(
+            self.base_url,
+            self.calendar_date.strftime('%m/%d/%Y')
+        )
+        return scrapy.Request(next_url, callback=self.parse)
+
+    def _parse_id(self, item):
+        """
+        Calulate ID. ID must be unique within the data source being scraped.
+
+        All links for PBCC appear to include eID or BID_ID, so raise an error
+        if not found rather than creating a slug.
+        """
+        event_link = item.css('a::attr(href)').extract_first()
+        if '?eID=' in event_link:
+            id_suffix = event_link.split('?eID=')[-1]
+        elif '?BID_ID=' in event_link:
+            id_suffix = event_link.split('?BID_ID=')[-1]
+        else:
+            raise ValueError('ID is required and not present in eID or BID_ID params')
+        return 'PBCC{}'.format(id_suffix)
+
+    def _parse_classification(self, item):
```

Comment:
I like using the PBCC event types for the classification. Agency's with a reasonable classification scheme mean we can filter easily on a combination of agency name and classification! 👍